agentlib/modules/simulator.py

```diff
@@ -406,18 +411,13 @@ def do_step(self):
         if not self.config.update_inputs_on_callback:
             # Update inputs manually
             self.update_model_inputs()
-        # Specify simulation end time
-        _end_time_simulation = self.env.time + self.config.t_sample
         # Simulate
         self.model.do_step(
             t_start=(self.env.now + self.env.offset + self.config.t_start),
             t_sample=self.config.t_sample
         )
         # Update the results and outputs
-        self._update_results(_end_time_simulation)
-        self.update_module_vars(timestamp=_end_time_simulation)
-        # Wait for the environment until the timeout
-        return self.env.timeout(self.config.t_sample)
+        self._update_results()
```

Comment:
They are different functions. One for outputs / states to broker, and one for outputs / states to results file.
I have to call results outputs once in the beginning to correctly create the first results row with the correct simulation start time. Because input writing always extends the existing row, while output writing opens the new row. To write inputs after the first step, I have to create this row before the first step.